luigi/contrib/ftp.py

```diff
@@ -131,14 +131,12 @@ def _sftp_exists(self, path, mtime):
         return exists
 
     def _ftp_exists(self, path, mtime):
-        path_parts = path.split('/')
-        path = '/'.join(path_parts[:-1])
-        fn = path_parts[-1]
+        dirname, fn = os.path.split(path)
 
-        files = self.conn.nlst(path)
+        files = self.conn.nlst(dirname)
 
         exists = False
-        if fn in files:
+        if path in files or fn in files:
```

Comment:
I tested that too. It takes 3x the time of just comparing twice against
each element. So it wouldn't make sense as an optimization unless we were
doing 7 or more comparisons.

On Fri, Jul 15, 2016 at 3:23 PM, Andy Sloane <notifications@github.com>
wrote:

> In luigi/contrib/ftp.py
> <https://github.com/spotify/luigi/pull/1751#discussion_r71048803>:
>
> >
> >          exists = False
> > -        if fn in files:
> > +        if path in files or fn in files:
>
> Or you could just change files into a set and then you have a more
> efficient test regardless of the circumstances
>
>     files = set(self.conn.nlst(dirname))
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spotify/luigi/pull/1751/files/9c0194e201782549b451ac02137bcae2f736add6#r71048803>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB_rbU-pXGEAFksTfWXCnwz20l_RtufRks5qWAhTgaJpZM4JGa9D>
> .
>